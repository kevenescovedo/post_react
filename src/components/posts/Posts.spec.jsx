import { render, screen } from "@testing-library/react"
import { Posts } from "."
const props = {
    posts: [{
        id: 1,
        title: 'title',
        body : 'body',
        cover: 'https://site.com',
    
    },
    {
        id: 2,
        title: 'title',
        body : 'body',
        cover: 'https://site2.com',
    
    }, 
    {
        id: 3,
        title: 'title',
        body : 'body',
        cover: 'https://site3.com',
    
    }
]
}
describe('</Posts>', () => {
    it('should render posts', () => {
      render(<Posts{...props}/>);
     //ve se tem postagem
     //ve se tem title
      expect(screen.getAllByRole('heading', {name: 'title' })).toHaveLength(3);
      //ve se tem img
      expect(screen.getAllByRole('img', {name: 'title' })).toHaveLength(3);
      //ve se tem paragrafo
      expect(screen.getAllByText("body")).toHaveLength(3);

    })

    it('should match snapshot', () => {
       const {container} = render(<Posts{...props}/>);
       expect(container.firstChild).toMatchSnapshot();
       
      })
})