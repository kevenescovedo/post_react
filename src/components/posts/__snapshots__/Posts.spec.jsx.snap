// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Posts> should match snapshot 1`] = `
<div
  class="container"
>
   
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="title"
        src="https://site.com"
      />
      <div
        class="post-content"
      >
        <h1>
          title
        </h1>
        <p>
          body
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title"
        src="https://site2.com"
      />
      <div
        class="post-content"
      >
        <h1>
          title
        </h1>
        <p>
          body
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="title"
        src="https://site3.com"
      />
      <div
        class="post-content"
      >
        <h1>
          title
        </h1>
        <p>
          body
        </p>
      </div>
    </div>
  </div>
</div>
`;
